src/libraries/Microsoft.PowerFx.Json/FormulaValueJSON.cs

```diff
@@ -57,18 +60,29 @@ public static FormulaValue FromJson(JsonElement element)
                     return new BooleanValue(IRContext.NotInSource(FormulaType.Boolean), true);
```

Comment:
Should these verify formulaType?
Ie, if FormulaType doesn't match the element.ValueKind?

Reply:
This could be related  to the new helpers in:
[[https://github.com/microsoft/Power-Fx/blob/main/src/libraries/Microsoft.PowerFx.Interpreter/Marshal/TypeCoercionProvider.cs](https://github.com/microsoft/Power-Fx/blob/main/src/libraries/Microsoft.PowerFx.Interpreter/Marshal/TypeCoercionProvider.cs)](https://github.com/microsoft/Power-Fx/blob/main/src/libraries/Microsoft.PowerFx.Interpreter/Marshal/TypeCoercionProvider.cs)